### PR TITLE
[IGA] Add function for communication with CL

### PR DESCRIPTION
**Pull request by tteschemacher**
This add functionalities required for a non linear CL for the shell 3p and does some minor cleanup

### CODE REVIEW

File: applications/IgaApplication/custom_elements/shell_3p_element.cpp [hunk 1]
```diff
@@ -79,15 +79,29 @@ namespace Kratos
...
+        // Reading integration points
+        const GeometryType& r_geometry = GetGeometry();
+        const Properties& r_properties = GetProperties();
+        const auto& N_values = r_geometry.ShapeFunctionsValues();
+
+        // Reading integration points
+        const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints();
+
+        for (IndexType point_number = 0; point_number < mConstitutiveLawVector.size(); ++point_number) {
+            mConstitutiveLawVector[point_number]->FinalizeSolutionStep(r_properties, r_geometry, row(N_values, point_number), rCurrentProcessInfo);
```
Comment (philbucher) [inline applications/IgaApplication/custom_elements/shell_3p_element.cpp]: This is long deprecated and removed in #7691 

please use the new method

Comment (tteschemacher) [reply applications/IgaApplication/custom_elements/shell_3p_element.cpp]: indeedily. Thanks!!

Whats about the initialize material, do we keep that beauty?

Comment (philbucher) [reply applications/IgaApplication/custom_elements/shell_3p_element.cpp]: Well you should not use the deprecated methods :)

Comment (tteschemacher) [reply applications/IgaApplication/custom_elements/shell_3p_element.cpp]: yeah but that one isn't deprecated [yet]

Comment (philbucher) [reply applications/IgaApplication/custom_elements/shell_3p_element.cpp]: I checked and I think there is no equivalent with `ConstitutiveLaw` parameters yet.
Hence I would leave it :)

**Review by philbucher — APPROVED** [review]

**Comment by tteschemacher** [comment]
🤙

**Pull request merged**